src/Symfony/Component/Translation/Bridge/Loco/LocoProvider.php

```diff
@@ -75,7 +75,12 @@ public function write(TranslatorBagInterface $translatorBag): void
             }
 
             foreach ($catalogue->all() as $domain => $messages) {
-                $ids = $this->getAssetsIds($domain);
+                $keysFromLoco = array_map(function ($id) use($domain) {
+                    return $this->retrieveKeyFromId($id, $domain);
+                }, $this->getAssetsIds($domain));
+
+                $ids = array_intersect($keysFromLoco, array_keys($messages));
```

Comment:
there's something strange from a logical pov:
the array combine below need the same cardinality in both argument
yet this intersection would mean that ids could have a lower cardinality
Do I miss something?

Reply:
You are right, I did something wrong here. I'm reworking on it in the coming days.